feat: add --resolve-image-repos support to sam package

abhishektang opened the pull request:
## Summary
Implements `--resolve-image-repos` support for `sam package` command to automatically create ECR repositories, matching the behavior of `sam deploy`.

## Resolves
Closes #3888

## Changes
- Added `--resolve-image-repos` CLI option to `sam package`
- Calls `sync_ecr_stack()` to auto-create managed ECR repositories
- Added validation requiring `--s3-bucket` when flag is used
- Added conflict detection with `--image-repositories`
- Added unit tests for validation logic

## Testing
- All existing tests pass
- Added new unit test for validation
- Tested with real AWS resources (ECR repos created successfully)

## Use Case
Enables CI/CD workflows to package once and deploy to multiple environments in parallel using AWS-managed ECR repositories, avoiding manual repo creation and permission complexity.

vicheey commented on samcli/commands/package/package_context.py:
```diff
@@ -101,6 +103,16 @@ def run(self):
         Execute packaging based on the argument provided by customers and samconfig.toml.
         """
+        if self.resolve_image_repos:
+            from samcli.lib.bootstrap.companion_stack.companion_stack_manager import sync_ecr_stack
```
Following the existing pattern, the import should be at the top of the file.

seshubaws commented:
Hi @abhishektang thanks for all your work on this! There seems to be a small linting error making some tests fail, could you check this?

abhishektang commented:
Hey, @seshubaws I have fixed the linting error, please try to run the PR workflow now.

abhishektang commented:
Hey @seshubaws @reedham-aws, I've fixed the unit test failures and added test coverage for the `_create_layer_directory` method that was missing from test runs.
